TrackingTools/GsfTools/interface/CloseComponentsMerger.icc

```diff
@@ -11,9 +15,118 @@ CloseComponentsMerger<N>::CloseComponentsMerger (int maxNumberOfComponents,
   theMaxNumberOfComponents(maxNumberOfComponents),
   theDistance(distance->clone()) {}
 
+
+
 template <unsigned int N> MultiGaussianState<N>
 CloseComponentsMerger<N>::merge (const MultiState& mgs) const
 {
+
+  //auto oldRes = mergeOld(mgs);
+
+  typedef std::vector<SingleStatePtr>  SingleStateVector;
+
+
+  auto ori = mgs.components();
+  SingleStateVector finalComponents;
+
+  int noComp = ori.size();
+  if (noComp <=theMaxNumberOfComponents) return mgs;
+
+
+  while (true) {
```

Comment:
It would be helpful to add a comment here explaining the conditions that guarantee that this loop will eventually terminate. Future developers would be thereby warned to keep those conditions in mind when modifying this code.